### PR TITLE
Closes #647 - Add Oracle and Worker to API

### LINKED ISSUE #647: Add Oracle and Worker to API
Inorder to easily support different methods of running Fluo (for example Mesos), it would be nice to add the Oracle to API.  Maybe something like the following :

```java
    FluoConfiguration conf = ...;
    Oracle oracle = FluoFactory.newOracle(conf);
    oracle.start();
    //wait for signal to shutdown
    oracle.stop();
```

Worker is too generic... thinking NotificationProcessor may be a better name to use in the API.

```java
   FluoConfiguration conf = ...;
   NotificationProcessor ntfyProc = FluoFactory.newNotificationProcessor(conf);
   ntfyProc.start();
   //wait for signal to shutdown
   ntfyProc.stop();
```

Could possibly implement [Guava Service](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/util/concurrent/Service.html)

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+package org.apache.fluo.api.service;
+
+/**
+ * Shared interface for Fluo services
+ */
+public interface FluoService {
+
+  /**
+   * Starts service
```
Comment (keith-turner) [inline modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java]: Should document expected behavior if service fails to start.  Should also document if this expected to be synchronous or async.

File: modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   */
+  void start();
+
+  /**
+   * Stops service cleanly
```
Comment (keith-turner) [inline modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java]: Should document expected behavior if unable to cleanly stop service.

Comment (keith-turner) [inline modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java]: My comment is influenced by the Java Iterator javadoc. That javadoc clearly defines what will happen in failure conditions.  This is useful for implementors and users.

File: modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java (hunk 1 above)
Comment (mikewalch) [inline modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java]: fixed in ad5f7602a0184ab

File: modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java (hunk 2 above)
Comment (mikewalch) [inline modules/api/src/main/java/org/apache/fluo/api/service/FluoService.java]: fixed in ad5f7602a0184ab

**Pull request closed without merging**